### ISSUE TITLE
IndexError: string index out of range

### ISSUE THREAD

**Issue by 1337-537h** [suurjaak/Skyperious#108]
Traceback (most recent call last):
			  File "skyperious\workers.py", line 276, in run
			  File "skyperious\skypedata.py", line 1849, in parse
			  File "skyperious\skypedata.py", line 2341, in dom_to_html
			  File "skyperious\workers.py", line 143, in <lambda>
			  File "wx\lib\wordwrap.py", line 36, in wordwrap
			IndexError: string index out of range

**Comment by suurjaak** [comment]
Few questions:

1. Does this happen for any text you enter in the search box? 
2. And in every chat you search in? 
3. How was Skyperious installed?

**Comment by 1337-537h** [comment]
1. No
2. Νο
3. I installed the latest Skyperious 5.1 x64 (not the portable).

**Comment by suurjaak** [comment]
What was the search query that produced the error?

**Comment by 1337-537h** [comment]
Greek letters/words
But in previous versions the problem wasn't there.

**Comment by suurjaak** [comment]
Can you give a specific instance of a query that raises error? Trying with Greek or Cyrillic or Kanji letters myself works successfully.

**Comment by 1337-537h** [comment]
First example: "νομιζω ειχα πεθάνει"

**Comment by suurjaak** [comment]
Does the message itself display fine in chat history, the message that contains this text? 

Can you give the contents of the matching message?

**Comment by 1337-537h** [comment]
The message appears itself fine in the history but not when im trying to search the keywords.

**Comment by suurjaak** [comment]
Can you give the contents of the matching message?

**Comment by 1337-537h** [comment]
no, can i help you diferently?

**Comment by suurjaak** [comment]
A round of 20 questions, then :) 


1. What happens when you search for any of those words individually? 
2. Where in the message text do these words occur? Beginning, end, in between? 
3. How long is the message itself? Is it an ordinary message? Are there some unusual characters in the message, neither Latin nor Greek alphabet? Is there some very long string of characters in the text? 
4. What happens when you search for something else that this message contains?

**Comment by 1337-537h** [comment]
everything is normal no big text in every line nothing special..
But the log (main.db) are big 200MB 300MB..
characters are greek and english (latin)
some searches return results some return the error i post above.

**Comment by suurjaak** [comment]
Can you respond to questions 1 and 4?

**Comment by 1337-537h** [comment]
1. If i search ειχα the error come up, but if i search "είχα" then no problem appears.
its not the same im just saying..
4.I thing always error again.

**Comment by suurjaak** [comment]
Difficult to track this problem down without specific information..

Can you try the following:

- open a database in Skyperious
- open the Python console (menu Help -> Show Python console)
- in the console, execute the following statements one by one:
```python
dc = wx.MemoryDC()

dc.SetFont(wx.Font(8, wx.FONTFAMILY_SWISS, wx.FONTSTYLE_NORMAL,
           wx.FONTWEIGHT_NORMAL, faceName=conf.HistoryFontName))

width = page.html_searchall.Size.width * 5 // 9

from wx.lib.wordwrap import wordwrap

```
And then see what are the results when you try invoking the following statement with different texts:
```python
wordwrap("YOURTEXT", width, dc)
```

**Comment by 1337-537h** [comment]
Is [that ](https://imgur.com/gY84tgw)correct?
error still there.

**Comment by suurjaak** [comment]
And now replace YOURTEXT with the contents of the message, whole and different parts of it, to see when and how the error occurs.

**Comment by 1337-537h** [comment]
Im not sure if i did it correctly.

`2022-05-03 21:51:32.914	An unexpected error has occurred:

			Traceback (most recent call last):
			  File "code.py", line 90, in runcode
			  File "<input>", line 1, in <module>
			NameError: name 'ειχα' is not defined
2022-05-03 21:52:23.848	An unexpected error has occurred:

			Traceback (most recent call last):
			  File "code.py", line 90, in runcode
			  File "<input>", line 1, in <module>
			NameError: name 'test' is not defined`

**Comment by suurjaak** [comment]
You need to put the text in quotes, like 
```python
wordwrap("ειχα", width, dc)
```

**Comment by suurjaak** [comment]
@1337-537h Are you still interested in getting this resolved?

**Comment by 1337-537h** [comment]
sure but i can do nothing.

**Comment by 1337-537h** [comment]
I think i did that when you told me and i reply you what happend.

**Comment by suurjaak** [comment]
In https://github.com/suurjaak/Skyperious/issues/108#issuecomment-1116556456 I gave further guidance on how to test the different parts of the problematic text.

**Comment by 1337-537h** [comment]
https://imgur.com/6zLvVa0

**Comment by suurjaak** [comment]
Can you identify a minimal text that raises this error, and that you are willing to share?

**Comment by 1337-537h** [comment]
I remember that happens when i search this word "ειχα"
But that appears in so many entries. The main.db file is over 300mb.
https://i.imgur.com/WjUGCTm.png

**Comment by suurjaak** [comment]
I think you misunderstood the guidance. After the `from wx.lib.wordwrap import wordwrap`, you should now write

```python
wordwrap("ειχα", width, dc)
```
Replacing `ειχα`  with the message text that is causing this error. Does it raise the error here also?

**Comment by 1337-537h** [comment]
same thing after that too.
https://i.imgur.com/BFtUP6h.png

**Comment by suurjaak** [comment]
Ok, how about you try this:

- open a database in Skyperious
- open the Python console (menu Help -> Show Python console)
- in the console, execute the following statements one by one:

```python

wordwrap_original = wx.lib.wordwrap.wordwrap

def wordwrap_logged(text, width, dc, breakLongWords=True, margin=0):
    try: return wordwrap_original(text, width, dc, breakLongWords, margin)
    except Exception as e:
        main.logger.info("ERROR WRAPPING %r: %s", text, e)
        raise

wx.lib.wordwrap.wordwrap = wordwrap_logged
```

And then try searching again with a text that raises error. This will print the exact problematic text to the Skyperious log window (menu Help -> Show log window).

**Comment by 1337-537h** [comment]
I didn't spot any difference..
https://i.imgur.com/ahbA8w2.png

**Comment by suurjaak** [comment]
You seemed to have missed pasting the last line:
```python
wx.lib.wordwrap.wordwrap = wordwrap_logged
```

**Comment by 1337-537h** [comment]
What you mean?
I did it, you can see it in the sreenshot.
I did it and its the same as the above screen.

In Python console i put:
1. wordwrap_original = wx.lib.wordwrap.wordwrap
2. def wordwrap_logged(text, width, dc, breakLongWords=True, margin=0):
    try: return wordwrap_original(text, width, dc, breakLongWords, margin)
    except Exception as e:
        main.logger.info("ERROR WRAPPING %r: %s", text, e)
        raise
3. wx.lib.wordwrap.wordwrap = wordwrap_logged
4.  search the keyword ειχα then the error come up and i open the log and take the screenshot.

**Comment by suurjaak** [comment]
Right, it's there, but it should be on a separate line - in the screenshot it has become part of the `wordwrap_logged` function definition.

**Comment by suurjaak** [comment]
Just copy-paste the blocks one by one, pressing Enter in between. First:

```python
wordwrap_original = wx.lib.wordwrap.wordwrap
```

Then:

```python
def wordwrap_logged(text, width, dc, breakLongWords=True, margin=0):
    try: return wordwrap_original(text, width, dc, breakLongWords, margin)
    except Exception as e:
        main.logger.info("ERROR WRAPPING %r: %s", text, e)
        raise
```

And then:

```python
wx.lib.wordwrap.wordwrap = wordwrap_logged
```

**Comment by 1337-537h** [comment]
https://i.imgur.com/CIz9ACo.png

**Comment by suurjaak** [comment]
Ok, so the error information gets logged now.

Can you show me the precise value that got ERROR WRAPPING?

**Comment by 1337-537h** [comment]
nooo.. its just a greek text.. is there any other way that can i help you?

**Comment by suurjaak** [comment]
Is all of it Greek text? If not, you can replace Greek letters with X.

**Comment by 1337-537h** [comment]
Thats another greek word that popup the same error that i can share the screenshot.
https://i.imgur.com/9VmA6xx.png

I search for the word "τεστ"  (test in greek)

**Comment by suurjaak** [comment]
Aha, got it. The problem comes up with text emoticons. 

Thank you, this gives me enough information to try fixing it.

**Comment by suurjaak** [comment]
Should be fixed in the newly released v5.2 - can you confirm?

**Comment by suurjaak** [comment]
Problem should be solved, no more feedback for 2 weeks - closing issue.
